openVfsFuse/src/openvfsfuse.cpp

```diff
@@ -58,19 +58,59 @@ using json = nlohmann::json;
 using namespace std;
 
 namespace {
-std::string mountPoint;
 
-std::string getAbsolutePath(const char *path)
+class VFSFuseContext
 {
-    return mountPoint + path;
-}
+public:
+    VFSFuseContext(const std::filesystem::path &mountPoint)
+        : _mountPoint(mountPoint)
+        , _rootHandle(open(_mountPoint.c_str(), 0))
+    {
+        assert(!_instance);
+        _instance = this;
+    }
+
+    ~VFSFuseContext() { close(_rootHandle); }
+
+    /***
+     * We are an overlay file system, using the same path internally and externally.
+     * To trick fuse and prevent it fromm intercepting our calls to the underlying file system,
+     * we chdir to the root and use relative paths internally.
+     */
+    auto getInternalPath(const std::string &path)
+    {
+        fchdir(_rootHandle);
+        return "." + path;
+    }
+
+    auto getExternalPath(const std::string &path) { return std::filesystem::path(_mountPoint.string() + path); }
+
+    auto mountPoint() const { return _mountPoint; }
+
+    static auto &instance() { return *_instance; }
+
+private:
+    static VFSFuseContext *_instance;
+    std::filesystem::path _mountPoint;
+    int _rootHandle;
+};
+
+VFSFuseContext *VFSFuseContext::_instance = nullptr;
 
+auto getInternalPath(const std::string &path)
```

Comment:
Hm, I find the rename to `getInternalPath` more confusing than letting it with `getRelativePath` - which it is.
Would also shrink this PR significantly ;-)